core/src/main/java/jenkins/I18n.java

```diff
@@ -106,8 +108,10 @@ public HttpResponse doResourceBundle(StaplerRequest request) {
             }
 
             return HttpResponses.okJSON(ResourceBundleUtil.getBundle(baseName, locale));
+        } catch (MissingResourceException e) {
+            return HttpResponses.errorJSON(e.getMessage(), HttpServletResponse.SC_NOT_FOUND);
         } catch (Exception e) {
-            return HttpResponses.errorJSON(e.getMessage());
+            return HttpResponses.errorJSON(e.getMessage(), HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```

Comment:
I see what you are doing here and it makes sense, but strictly speaking is not needed for fixing the issue wrt loading bundles from plugins, or am I missing something?